controllers/datanetwork_controller.go

```diff
@@ -537,7 +537,14 @@ func (r *DataNetworkReconciler) Reconcile(ctx context.Context, request ctrl.Requ
 	}
 
 	if instance.Status.ObservedGeneration == instance.ObjectMeta.Generation && instance.Status.Reconciled {
-		return ctrl.Result{}, nil
+		// Update the deploymentScope if necessary when Generation is same
+		// as ObservedGeneration, instance is reconciled.
+		err, scope_updated := r.UpdateDeploymentScope(instance)
+		if err != nil {
+			return reconcile.Result{}, err
+		} else if !scope_updated {
+			return ctrl.Result{}, nil
+		}
 	}
 
 	// Update scope from configuration
```

Comment:
Expect to update the comment/log/method description around the UpdateConfigStatus. And if the update scope is done in UpdateDeploymentScope, it should not done again in UpdateConfigStatus

Reply:
Done, updated method description around UpdateConfigStatus across all controllers. Scope is either updated in UpdateDeploymentScope or UpdateConfigStatus (among other status parameters in this case) depending on Generation v/s ObservedGeneration comparison but will it be updated twice.

Reply:
Doesn't make sense to have two methods to update scope twice, can you update the logic to avoid that?

Reply:
Done, separated "deploymentScope" update logic from "UpdateConfigStatus" function.